public/models/GameMap.js

```diff
@@ -30,105 +30,117 @@ class GameMap {
+            path.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            // take one coin to draw as they are the same
+            coin.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          case 1:
+            wall.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          case 2:
+            goal.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          default:
+        }
+      }
     }
+  }
 
-    //Marianna
-    //load map from the file
-    //set goal tile and time limit
-    loadMap (data) {
-        this.tiles = data.tiles;
-        //get random position for goal
-        let [row, column] = [-1, -1];
-        do {
-            row = Utils.getRandomNumber(0, this.tiles.length);
-            column = Utils.getRandomNumber(0, this.tiles[row].length);
-        } while (this.tiles[row][column] !== 0);
-        this.tiles[row][column] = 2;
-        this.goalRow = row;
-        this.goalColumn = column;
-        //set time limit based on size - 1 minute for 10x10
-        this.timeLimit = 60 * 1000 * (this.tiles.length * this.tiles[0].length / 100);
-        //generate coins
-        //5 coins per 10x10 
-        this.generateCoins(this.tiles.length * this.tiles[0].length / 20)
-    }
+  // Marianna
+  // load map from the file
+  // set goal tile and time limit
+  loadMap(mapFile) {
+    const fs = require('fs');
+    const data = fs.readFileSync(mapFile);
+    this.tiles = JSON.parse(data).tiles;
+    // get random position for goal
+    let [row, column] = [-1, -1];
+    do {
+      row = Utils.getRandomNumber(0, this.tiles.length);
+      column = Utils.getRandomNumber(0, this.tiles[row].length);
+    } while (this.tiles[row][column] !== 0);
+    this.tiles[row][column] = 2;
+    this.goalRow = row;
+    this.goalColumn = column;
+    // set time limit based on size - 1 minute for 10x10
+    this.timeLimit = 60 * 1000 * (this.tiles.length * this.tiles[0].length / 100);
```

Comment:
1000 what is this number?

Reply:
time is in miliseconds so to get 1 minute i need to do *1000 which will be 1 second and then *60 which will be 60 seconds aka 1 minute